### PR TITLE
add progress barz

### PR DESCRIPTION
**Pull request by oodtoon**

### CODE REVIEW

File: src/lib/components/ProgressBar.svelte [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	</div>
+	<div class="progress-bar-outer">
+    <div class="progress-bar-inner" style="width: {$progress * 100}%; background-color: {color}" ></div>
+  </div>
+</div>
+
+<style>
+	.progress-container {
+		display: grid;
+		grid-template-columns: 1fr 9fr;
```
Comment (dgore7) [inline src/lib/components/ProgressBar.svelte]:
```suggestion
		display: flex;
```

File: src/lib/components/ProgressBar.svelte [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    justify-self: start;
+    display: flex;
+    border-radius: 10px;
+    height: 18px;
+  }
+
+	.progress-bar-outer {
+		background-color: white;
+    width: 90%;
+	}
```
Comment (dgore7) [inline src/lib/components/ProgressBar.svelte]: simplification - `overflow: hidden` will prevent inner from escaping the bounds and remove the need for `border-radius` on inner
```suggestion
	.progress-bar-outer {
		border-radius: 10px;
		height: 18px;
		display: flex;
		overflow: hidden;
		background-color: white;
	}
	}```

File: src/lib/components/ProgressBar.svelte [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+<style>
+	.progress-container {
+		display: grid;
+		grid-template-columns: 1fr 9fr;
+		align-items: center;
+	}
+
+  .player-name {
+    margin: auto;
+  }
```
Comment (dgore7) [inline src/lib/components/ProgressBar.svelte]: appears unneeded. Looks the same for small screens. For larger screens, we can use margin on the `main` tag to create horizontal space on both ends

File: src/lib/components/ProgressBar.svelte [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+	export let player: Player;
+  export let color: string;
+
+  const progress = tweened(0, {
+    duration: 500,
+    easing: cubicOut,
+  })
+
+  $: progressPercent = player.score / player.scoreRequired
+  $: progress.set(progressPercent)
```
Comment (dgore7) [inline src/lib/components/ProgressBar.svelte]: nice use of tweened!

**Review by dgore7 — COMMENTED** [review]
Don't forget to format, and make sure you take care of warnings
Screenshot 2024-04-22 at 9 27 13 AM

File: src/lib/components/ProgressBar.svelte (hunk 2 above)
Comment (oodtoon) [reply src/lib/components/ProgressBar.svelte]: The bar looks better with a radius imo. I want to leave it. It's too jarring with it being flat.

**Comment by oodtoon** [comment]
The formatter wasn't formatted when I pushed this. You fixed the formatter when I showed you locally on this mac.

File: src/lib/components/ProgressBar.svelte (hunk 2 above)
Comment (dgore7) [reply src/lib/components/ProgressBar.svelte]: It shouldn't be flat. This is purely a refactor - meaning it should be functionally the same

**Pull request merged**